appservice/src/github/connectToGitHub.ts

```diff
@@ -7,29 +7,35 @@ import { SiteSourceControl } from 'azure-arm-website/lib/models';
 import { TokenCredentials, WebResource } from 'ms-rest';
 import { Response } from 'request';
 import * as request from 'request-promise';
+import { isArray } from 'util';
 import * as vscode from 'vscode';
-import { appendExtensionUserAgent, AzureTreeItem, DialogResponses, IActionContext, IAzureQuickPickItem, IParsedError, openInPortal, parseError } from 'vscode-azureextensionui';
-import { ext } from './extensionVariables';
-import { localize } from './localize';
-import { signRequest } from './signRequest';
-import { SiteClient } from './SiteClient';
-import { nonNullProp } from './utils/nonNull';
-import { openUrl } from './utils/openUrl';
-import { verifyNoRunFromPackageSetting } from './verifyNoRunFromPackageSetting';
-
-type gitHubOrgData = { login: string, repos_url: string };
-type gitHubReposData = { name: string, repos_url: string, url: string, html_url: string };
-type gitHubBranchData = { name: string };
-type gitHubLink = { prev?: string, next?: string, last?: string, first?: string };
+import { appendExtensionUserAgent, AzureTreeItem, AzureWizard, DialogResponses, IActionContext, IAzureQuickPickItem, IParsedError, openInPortal, parseError } from 'vscode-azureextensionui';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { signRequest } from '../signRequest';
+import { SiteClient } from '../SiteClient';
+import { nonNullProp } from '../utils/nonNull';
+import { openUrl } from '../utils/openUrl';
+import { verifyNoRunFromPackageSetting } from '../verifyNoRunFromPackageSetting';
+import { GitHubBranchListStep } from './GitHubBranchListStep';
+import { GitHubOrgListStep } from './GitHubOrgListStep';
+import { GitHubRepoListStep } from './GitHubRepoListStep';
+import { IConnectToGitHubWizardContext } from './IConnectToGitHubWizardContext';
+
+export type gitHubOrgData = { login: string, repos_url: string };
+export type gitHubRepoData = { name: string, repos_url: string, url: string, html_url: string };
+export type gitHubBranchData = { name: string };
+export type gitHubLink = { prev?: string, next?: string, last?: string, first?: string };
 // tslint:disable-next-line:no-reserved-keywords
-type gitHubWebResource = WebResource & { resolveWithFullResponse?: boolean, nextLink?: string, type?: string };
+export type gitHubWebResource = WebResource & { resolveWithFullResponse?: boolean, nextLink?: string, type?: string };
 
-export async function connectToGitHub(node: AzureTreeItem, client: SiteClient, context: IActionContext): Promise<void> {
-    const requestOptions: gitHubWebResource = new WebResource();
-    requestOptions.resolveWithFullResponse = true;
-    requestOptions.headers = {
+export async function connectToGitHub(node: AzureTreeItem, client: SiteClient, context: IConnectToGitHubWizardContext): Promise<void> {
+    context.requestOption = new WebResource();
```

Comment:
Not a huge fan of re-using the same requestOption. Can you create a new `requestOption` inside `getGitHubJsonResponse` and take the url as a parameter instead?

Specifically worried about one of the requests setting a header or something and other requests inheriting that on accident